.github/pull_request_template.md

```diff
@@ -17,3 +17,4 @@
 * [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
 * [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
 * [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
+* [ ] I am referencing an issue in this PR or in the commit message for auto cleanup. For example, `fixes: #1234` or `closes: #567`.
```

Comment:
__If a PR resolves an issue then__ it should contain a link to that issue.

I do not want to mandate that all PRs _need_ an issue first.  I find that to be an inefficient approach.

Reply:
See the commit message above... `Resolves:` also works.